urpc/urpc.go

```diff
@@ -18,14 +18,11 @@ func (srv *Server) rpcHandle(conn grove_ffi.Connection, rpcid uint64, seqno uint
 	f := srv.handlers[rpcid] // for Goose
 	f(data, replyData)       // call the function
 
-	// assume no overflow (*replyData would have to be almost 2^64 bytes large...)
-	num_bytes := std.SumAssumeNoOverflow(8+8, uint64(len(*replyData)))
-	e := marshal.NewEnc(num_bytes)
-	e.PutInt(seqno)
-	e.PutInt(uint64(len(*replyData)))
-	e.PutBytes(*replyData)
+	data1 := make([]byte, 0, 8+len(*replyData))
+	data2 := marshal.WriteInt(data1, seqno)
```

Comment:
Is there a reason to not just write `data := _; data := _; ...` and rely on shadowing? I don't know how that would look in Coq.

Reply:
Yeah the reason is that Go rejects such code. :(

Reply:
You can actually do this if you create a new scope:

```go
func foo(b []byte) int {
	{
		b := 3
		return b
	}
}
```

This is probably a bad plan in conjunction with Goose, though. `a, b := ...` is also tricky when it assigns to b and defines `a`.

Reply:
Also it will look terrible because of all the nested blocks.

> a, b := ... is also tricky when it assigns to b and defines a.

I thought it would define both and do shadowing? I am using that for the `ReadInt` code above.